libs/sdk-core/src/breez_services.rs

```diff
@@ -235,6 +235,23 @@ impl BreezServices {
     ) -> SdkResult<Payment> {
         self.start_node().await?;
         let parsed_invoice = parse_invoice(bolt11.as_str())?;
+        let invoice_amount_msat = parsed_invoice.amount_msat.unwrap_or_default();
+        let provided_amount_msat = amount_sats.unwrap_or_default() * 1000;
+
+        // Ensure amount is provided for zero invoice
+        if provided_amount_msat == 0 && invoice_amount_msat == 0 {
+            return Err(SdkError::SendPaymentFailed {
+                err: "amount must be provided when paying a zero invoice".into(),
+            });
+        }
+
+        // Ensure amount is not provided for invoice that contains amount
+        if provided_amount_msat > 0 && invoice_amount_msat > 0 {
```

Comment:
Should we consider allowing if non-zero amounts match?

Reply:
I think it is better to be strict here. Greenlight doesn't allow it even if matches.

Reply:
I think we should. If the amounts match, we should ignore the provided amount. Because the method signature takes an amount parameter, you expect you have to pass an amount there. Developers will have already parsed the invoice to show the amount to the user. So they already have the amount in a variable. It makes sense to pass it here. But that only really works if the input parameter is in millisatoshi, otherwise it only works full full satoshi amounts.

Reply:
Ah, Roei's comment wasn't there yet when I answered. No strong opinion tbh.

Reply:
The fact the amount is parsed in the invoice doesn't necessary mean it is required in the `send_payment` and the signature takes optional amount.
Don't you think it feels weird to allow passing a parameter only in the case it is actually redundant? I mean how would we document such behavior? "Amount is allowed if it is the same as in the invoice"?
It would make sense IMO if the node supports passing amount that is greater or equal to the invoice amount but this is not the case here.
Isn't it more coherent (and clear to the user) behavior to require (and allow) amount only in case of zero invoice and avoid this redundancy?
What do you think?

Reply:
@JssDWt comments wasn't there when I answered :)
Anyway let me know your final thoughts.

Reply:
@roeierez Yeah makes sense. We were getting quite a few questions about this particular issue. I think the error is much better now and should probably lead to less questions already. Maybe make it clearer in the docs too.

Reply:
How about supporting both with `SendPaymentRequest` and `SendPaymentZeroAmountInvoice`?

Reply:
Nevermind, that would complicate the service interface for very little benefit.

I agree with @JssDWt above.